main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -609,122 +582,34 @@ object SetFormula {
     * Nested unions are put into a single union.
     */
   def mkUnionAll(fs: List[SetFormula]): SetFormula = {
-    // We need to do two things:
-    // - Separate subformulas into specific buckets of pos/neg elements/variables/constants and other.
-    // - Flatten nested unions.
+    def visit(l: List[SetFormula], elmAcc: SortedSet[Int], seenCsts: Set[Int], seenVars: Set[Int]): List[SetFormula] = l match {
```

Comment:
Shouldn't these two be `SortedSet`s?

Reply:
I made them into SortedSet. They are only used for lookups (never traversal) so order shouldn't matter.